lib/internal/modules/cjs/loader.js

```diff
@@ -567,7 +569,16 @@ function tryModuleLoad(module, filename) {
 }
 
 Module._resolveFilename = function(request, parent, isMain, options) {
-  if (NativeModule.nonInternalExists(request)) {
+  if (request.search('@nodejs/') === 0) {
```

Comment:
This is where we could add a check to ensure that only new modules are being loaded from namespace